datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -688,6 +688,29 @@ fn build_predicate_expression(
                 return Ok(unhandled);
             }
         }
+        Expr::InList {
+            expr,
+            list,
+            negated,
+        } => {
+            if !list.is_empty() {
```

Comment:
This check could be added to the InList match variant instead

Reply:
😂  i don't understand this one, how should i achieve this? maybe i lost something in learning rust😔

Reply:
You can add the condition after the pattern

`(inlist pattern) if !list.is_empty() => {...}`



See the docs here: https://doc.rust-lang.org/rust-by-example/flow_control/match/guard.html

Reply:
Thanks @Dandandan ❤️